fitapp/tasks.py

```diff
@@ -46,24 +46,22 @@ def unsubscribe(*args, **kwargs):
         raise Reject(exc, requeue=False)
 
 
-
 @shared_task
-def get_time_series_data(fitbit_user, cat, resource, date=None):
+def get_time_series_data(fitbit_user, categories=[], date=None):
     """ Get the user's time series data """
 
-    try:
-        _type = TimeSeriesDataType.objects.get(category=cat, resource=resource)
-    except TimeSeriesDataType.DoesNotExist:
-        logger.exception("The resource %s in category %s doesn't exist" % (
-            resource, cat))
+    filters = {'category__in': categories} if categories else {}
+    types = TimeSeriesDataType.objects.filter(**filters)
+    if not types.exists():
+        logger.exception("Couldn't find the time series data types")
         raise Reject(sys.exc_info()[1], requeue=False)
 
     # Create a lock so we don't try to run the same task multiple times
     sdat = date.strftime('%Y-%m-%d') if date else 'ALL'
-    lock_id = '{0}-lock-{1}-{2}-{3}'.format(__name__, fitbit_user, _type, sdat)
+    cats = '-'.join('%s' % i for i in categories)
+    lock_id = '{0}-lock-{1}-{2}-{3}'.format(__name__, fitbit_user, cats, sdat)
     if not cache.add(lock_id, 'true', LOCK_EXPIRE):
```

Comment:
@grokcode What can I do then? Would it be safe to get rid of this lock and decorate `get_fitbit_data` with `@transaction.atomic()`?